crates/polars-plan/src/plans/optimizer/simplify_expr.rs

```diff
@@ -440,6 +440,63 @@ impl OptimizationRule for SimplifyExprRule {
         let expr = expr_arena.get(expr_node).clone();
 
         let out = match &expr {
+            // drop_nulls().len() -> null_count()
+            // drop_nulls().count() -> null_count()
+            AExpr::Agg(IRAggExpr::Count(input, _)) => {
+                let input_expr = expr_arena.get(*input);
+                match input_expr {
+                    AExpr::Function {
+                        input,
```

Comment:
For all of these optimizations, can you explicitly ensure that the input is a column during match? If the input is a long chain of expressions and CSE isn't enabled, we could end up with perf regressions.

Reply:
Correct me if I am wrong - I need to see that the 'input' is a a vector with a single element, and in this case take its AExpr and see if it matches Column expression, am I right?

Reply:
It should be - you can give it a try to see if it works

Reply:
Works great, I will finish other tiny changes and will push all fixes for review.